index.html

```diff
@@ -167,42 +168,54 @@ <h3 class="sr-only">Math editor</h3>
-                                        title="Clean up the cross outs and start a new step (⌨: shift+enter)"
-                                        onclick="AddStep();"
-                                        data-step="3"
-                                        data-intro="Clean-up your work and start a new step."><i
-                                        class="fa fa-plus"></i> Add Step
-                                </button>
-                                <div hidden id="updateControls">     
-                                    <button id='updateStep' class="btn btn-success pointer editorButton" data-toggle="tooltip"
-                                        title="Update the step (⌨: shift+enter)"
-                                        data-step="3"
-                                        data-intro="Update the step."><i
-                                        class="fa fa-pencil"></i> Update Step
-                                    </button>                         
-                                    <br/>
-                                    <button id='cancelEdit' class="btn btn-default pointer editorButton" data-toggle="tooltip"
-                                        title="Cancel edit" onclick="ExitUpdate();"
-                                        data-step="3"
-                                        data-intro="Cancel edit."><i
-                                        class="fa fa-ban" ></i> Cancel edit
-                                    </button>                        
+                            <div id="mathPalette" class="pl-0 d-flex flex-nowrap"></div>
+                            <div id="mathButtons" class="d-flex flex-nowrap justify-content-between">
+                                <div>
+                                    <br>
+                                    <button id="scratch-pad-button" class="btn btn-default btn-scratch btn-background paletteButton" data-toggle="tooltip" title="Display/hide sketch pad">
+                                    </button>
+                                </div>
+                                <div>
+                                    <button class="btn btn-undo btn-background paletteButton" id="undoDelete" data-toggle="tooltip"
+                                            title="Undo Last Delete" onclick="UndoDeleteStep()" style="margin-bottom: 5px;"></button>
+                                    <br>
+                                    <h3 class="sr-only">Clean up and add new step</h3>
+                                    <button id='addStep' class="btn btn-primary pointer editorButton" data-toggle="tooltip"
+                                            title="Clean up the cross outs and start a new step (⌨: shift+enter)"
+                                            onclick="AddStep();"
+                                            data-step="3"
+                                            data-intro="Clean-up your work and start a new step."><i
```

Comment:
Please move this mark to the new line.

Reply:
Right, I have also noticed that there is used an improper markup <i>. I will change it to <span>